### ISSUE TITLE
Updating dynamically the background-image for ngStyle variable doesn't work in Safari or IE.

### ISSUE THREAD

**Issue by samuq** [angular/angular#24685]
This bug was spotted in Angular 5.

## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x ] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior

If you try to update background-image for ngSTyle variable ([ngStyle]="variable") from within component (this.variable={'background-image': 'url('+file.url + ')'};) it works only once (on ngOnInit) but doesn't update dynamically afterwards in Safari and IE. It does work on Chrome and Firefox.
Following assignment seems to work in all browsers tested: this.variable={'background': 'url('+file.url + ')'};

## Expected behavior
Background-image property would change dynamically.

## Reproduce error
https://stackblitz.com/edit/angular-euxy4c

## Environment

<pre><code>
Angular version: 5

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [x] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [x] IE version XX
- [ ] Edge version XX

**Comment by pkozlowski-opensource** [comment]
Could you please provide a minimal reproduce scenario in stackblitz / plunker?

**Comment by pkozlowski-opensource** [comment]
Can't reproduce on Safari Version 11.1.1 (12605.2.8) (OSX) with: https://stackblitz.com/edit/angular-5jgb7x?file=src%2Fapp%2Fapp.component.ts

**Comment by samuq** [comment]
Here's reproduced scenario: https://stackblitz.com/edit/angular-try63k?embed=1&file=src/app/image-carousel.component.ts

**Comment by pkozlowski-opensource** [comment]
@samuq there is a _lot_ going on in your stackblitz _plus_ there are some errors in the console. I'm sorry but you will have to try to simplify your reproduction scenario to the bare minimum that shows the issue. In your current stackblitz there is nothing that would suggest issue being in Angular / `ngStyle` - in fact I'm suspecting an issue in your CSS...

Sorry, we need more isolated scenario that clearly shows issue in Angular...

**Comment by samuq** [comment]
OK, now there's no errors in the console: https://angular-try63k.stackblitz.io/ 
and it works on Firefox + Chrome, so it's not an error with CSS. The background-image just isn't changed in safari + IE. It works if you replace 'background-image' with 'background'.

**Comment by samuq** [comment]
@pkozlowski I think I can’t minimize the code, because the error seems to occur in that specific context.

**Comment by samuq** [comment]
This works in safari as well: https://stackblitz.com/edit/angular-try63k . Seems like you can't set background-image for Safari and IE from within AnimationPlayer.onDone-function.

**Comment by samuq** [comment]
Reproduce error: https://stackblitz.com/edit/angular-euxy4c. 
Working version: https://stackblitz.com/edit/angular-try63k

**Comment by petebacondarwin** [comment]
Sorry for the silence on this issue. I tried out the reproductions. For me it now works with Safari as well. So I suspect that it was a browser bug rather than a bug in the framework.

I can see that it is still an issue in IE.

Given that there is a simple workaround and it only affects IE, I propose that we document this as a known issue with a workaround.

**Comment by jessicajaniuk** [comment]
Thanks for reporting this issue. This issue is now obsolete due to changes in the recent releases. Please update to the most recent Angular version.

If the problem still exists in your application, please [open a new issue](https://github.com/angular/angular/issues/new/choose) and follow the instructions in the issue template.
